src/npctrade.cpp

```diff
@@ -711,13 +713,13 @@ bool trading_window::perform_trade( npc &np, const std::string &deal )
                 popup( _( "%s can't carry all that." ), np.get_name() );
             } else if( np.mission != NPC_MISSION_SHOPKEEP && !npc_can_fit_items( np ) ) {
                 popup( _( "%s doesn't have the appropriate pockets to accept that." ), np.get_name() );
+            } else if( np.mission == NPC_MISSION_SHOPKEEP && !npc_can_trade_back( np ) ) {
+                popup( _( "%s don't trust you enough to sell you that." ), np.get_faction()->name );
```

Comment:
This check should rather be done in `npc::wants_to_sell()`

Reply:
Cheers, I moved this check and allowed negative trust values as you suggested.